src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -1241,34 +1236,41 @@ TU_ATTR_WEAK bool dcd_alloc_mem_for_conf(uint8_t rhport, tusb_desc_configuration
 
   // Determine number of used out EPs of current configuration and size of two biggest out EPs
   uint8_t nUsedOutEPs = 0, cnt_ep, cnt_tt;
-  bool tmp;
   uint16_t sz[2] = {0, 0};
 
   for (cnt_ep = 0; cnt_ep < EP_MAX; cnt_ep++)
   {
-    tmp = false;
     for (cnt_tt = 0; cnt_tt <= TUSB_XFER_INTERRUPT; cnt_tt++)
     {
-      tmp |= report.ep_transfer_type[cnt_ep][TUSB_DIR_OUT][cnt_tt];
+      if (report.ep_transfer_type[cnt_ep][TUSB_DIR_OUT][cnt_tt])
+      {
+        nUsedOutEPs++;
+        break;
+      }
     }
-    nUsedOutEPs += tmp;
 
     if (sz[0] < report.ep_size[cnt_ep][TUSB_DIR_OUT])
     {
       sz[1] = sz[0];
       sz[0] = report.ep_size[cnt_ep][TUSB_DIR_OUT];
     }
+    else if (sz[1] < report.ep_size[cnt_ep][TUSB_DIR_OUT])
+    {
+      sz[1] = report.ep_size[cnt_ep][TUSB_DIR_OUT];
+    }
   }
 
   // For configuration use the approach as explained in bus_reset()
-  _allocated_fifo_words = 15 + 2*nUsedOutEPs + (sz[0] / 4) + (sz[0] % 4 > 0 ? 1 : 0) + (sz[1] / 4) + (sz[1] % 4 > 0 ? 1 : 0) + 2;   // again, i do not really know why we need + 2 but otherwise it does not work
+  _allocated_fifo_words = 13 + 1 + 1 + 2 * nUsedOutEPs + ((sz[0] / 4) + (sz[1] / 4) + 1) + 2;   // again, i do not really know why we need + 2 but otherwise it does not work
 
   usb_otg->GRXFSIZ = _allocated_fifo_words;
 
   // Control IN uses FIFO 0 with report.ep_size[0][TUSB_DIR_IN] bytes ( report.ep_size[0][TUSB_DIR_IN]/4 32-bit word )
-  usb_otg->DIEPTXF0_HNPTXFSIZ = (report.ep_size[0][TUSB_DIR_IN]/4 << USB_OTG_TX0FD_Pos) | _allocated_fifo_words;
+  uint16_t fifo_depth = report.ep_size[0][TUSB_DIR_IN] / 4;
+  fifo_depth = tu_max16(16, fifo_depth);
```

Comment:
Why is the minimum 16? I think i tested this version with every EP0 size (down to 8 bytes) and it worked!

Reply:
All TX FIFO (**OTG_FS_DIEPTXFx**,**OTG_FS_HPTXFSIZ**, **OTG_FS_DIEPTXF0**) registers are marked with comment '`Minimum value is 16`'. I've never checked if it will work otherwise or not, if it would not work for some MCU it could be hard to track it think.
This applies to F4, L4 MCUs.

Reply:
Oh wow i did not see that! Thanks!